### PR TITLE
[Detail API] fix for deleting old index value, to update new index value

### PR DESCRIPTION
**Pull request by sujatha-mh**
Issue: https://github.com/egraphdb/egraphdb/issues/2

### CODE REVIEW

File: models/egraph_detail_model.erl [hunk 1]
```diff
@@ -754,7 +754,7 @@ change_indexes(Type, RawKey, OldIndexes, Indexes, DbStoredDetailsInfo, UpdatedDe
     DeletedIndexResults = delete_index(
                             ChangedIndexes ++ DeletedIndexes,
                             Type,
-                            DbStoredDetailsInfo,
+                            DbStoredDetailsInfo, RawKey,
```
Comment (neeraj9) [inline models/egraph_detail_model.erl]: Add "RawKey" in separate line.

File: models/egraph_index_model.erl [hunk 2]
```diff
@@ -242,34 +242,49 @@ create_or_update_info(Info, State) ->
     end.
 
 delete_resource(Key, KeyType, IndexName) ->
+    {Query, Params} = form_delete_index_base_query(Key, KeyType, IndexName),
```
Comment (neeraj9) [inline models/egraph_index_model.erl:245]: You don't have to duplicate so much of code. Instead you could just write a new function to generate query and params instead and reuse most of it there.

```
delete_resource(Key, KeyType, IndexName) ->
    {Query, Params} = form_delete_index_base_query(Key, KeyType, IndexName),
    delete_resource_internal(Query, Params).

delete_resource(Key, KeyType, Value, IndexName) ->
    {Query, Params} = form_delete_index_base_query(Key, KeyType, IndexName),
    Query1 = <<Query/binary, " and id=?">>,
    Params1 = Params ++ [Value],
    delete_resource_internal(Query1, Params1).

delete_resource_internal(Query, Params) ->
    TimeoutMsec = egraph_config_util:mysql_rw_timeout_msec(index),
    PoolName = egraph_config_util:mysql_rw_pool(index),
    case egraph_sql_util:mysql_write_query(
        PoolName, Query, Params, TimeoutMsec) of
        ok ->
            true;
        _ ->
            false
    end.
```

File: models/egraph_index_model.erl [hunk 3]
```diff
@@ -242,34 +242,49 @@ create_or_update_info(Info, State) ->
...
+    case egraph_sql_util:mysql_write_query(
+           PoolName, Query, Params, TimeoutMsec) of
+        ok ->
+            true;
+        _ ->
+            false
+    end.
+delete_resource(Key, KeyType, Value, IndexName) ->
+    {Query, Params} = form_delete_index_base_query(Key, KeyType, IndexName),
+    Query1 = iolist_to_binary([Query] ++ " and id=?"),
```
Comment (neeraj9) [inline models/egraph_index_model.erl]: Ideally that should be Query1 = <<Query/binary, " and id=?">>,

**Review by neeraj9 — CHANGES_REQUESTED** [review]
please close these before it can be merged with the rest of the codebase.

**Comment by sujatha-mh** [comment]
I have made the changes as suggested. Please validate and accept this pull request.

**Comment by neeraj9** [comment]
Looks good now. Thanks.

**Pull request merged**